serde-generate/runtime/typescript/serde/binarySerializer.ts

```diff
@@ -84,7 +84,7 @@ export abstract class BinarySerializer implements Serializer {
         this.serializeWithFunction(DataView.prototype.setUint32, 4, value);
     }
 
-    public serializeU64(value: BigInt | number): void {
```

Comment:
I don't understand this change. The JS community seems to be using `BigInt`.

Reply:
For the constructor. Every primitive in JS has a constructor and the type, which typically are only differentiated by capitalization. In JS world this doesn't matter, in Typescript it does - they are different types, one is a function and one is the actual value. This PR and the feature involved is about Typescript and not JS, and Typescript does not compile with this currently.

See that the code uses `number` and not `Number`, or `string` and not `String`. It's the same.

If you check the official documentation for BigInt in typescript, this will show that `bigint` is used as the type of a value that is big int, and `BigInt` is used as a function to convert an input into `bigint`. In Rust terms this is `Foo` vs `Foo::new()`.
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-2.html#bigint

```typescript
let foo: bigint = BigInt(100); // the BigInt function
let bar: bigint = 100n; // a BigInt literal
```